### PR TITLE
Countdown

### PR DESCRIPTION
**Pull request by nilshindermann**
# Dynamic Countdown page

This PR adds the `/countdown/[event]` dynamic page

## Valid event slugs
- `new-year`
- `may-the-fourth`
- `nils`
- `august`
- `x-mas`
- `Y2K38`

### CODE REVIEW

**Review by llettieri — DISMISSED** [review]

**Comment by llettieri** [comment]
[like]  Lettieri, Lorenzo reacted to your message:
________________________________

File: app/countdown/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    title: '404 - Nils Hindermann',
+    description: 'Das gewünschte Ereignis konnte nicht gefunden werden.',
+};
+
+export default function EventNotFoundPage(): ReactElement {
+    return (
+        <>
+            <h1 className="mb-10 text-6xl font-bold">404</h1>
+            <p className="text-xl">
+                Leider wissen wir nicht, welches Ereignis du meinst.
```
Comment (llettieri) [inline app/countdown/not-found.tsx]: Who's this?

File: components/countdown/Countdown.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    minutes: number;
+    seconds: number;
+}
+
+export default function Countdown({
+    event,
+}: {
+    event: EventDate;
+}): ReactElement {
+    const calcTime = useCallback((): Time => {
```
Comment (llettieri) [inline components/countdown/Countdown.tsx]: Maybe consider this lib https://day.js.org/

File: components/countdown/EventSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    current?: CountdownEvent;
+    basePath?: string;
+}
+
+export default function EventSelect({
+    events,
+    current,
+    basePath,
+}: Props): ReactElement {
+    const router = useRouter();
```
Comment (llettieri) [inline components/countdown/EventSelect.tsx]: Good one!

File: components/countdown/EventSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            </div>
+            <ul
+                tabIndex={0}
+                className="menu dropdown-content z-[1] w-52 rounded-box bg-neutral p-2 shadow"
+            >
+                {events
+                    .filter((e) => !current || e.slug !== current.slug)
+                    .map((e) => (
+                        <li key={e.slug}>
+                            <a onClick={() => router.push(`${path}/${e.slug}`)}>
```
Comment (llettieri) [inline components/countdown/EventSelect.tsx]: What about the Link component from Next.js?

File: components/countdown/EventSelect.tsx (hunk 3 above)
Comment (nilshindermann) [reply components/countdown/EventSelect.tsx]: What do you mean?

File: components/countdown/Countdown.tsx (hunk 2 above)
Comment (nilshindermann) [reply components/countdown/Countdown.tsx]: > Maybe consider this lib https://day.js.org/

Do you have any tips? Or can I just leave it as it is for now?

File: models/EventDate.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+dayjs.extend(utc);
+
+export default interface EventDate {
+    day: number;
+    month: number;
+    year?: number;
+}
+
+export const getDateFromEvent = (eventDate: EventDate): Date => {
+    if (eventDate.year) {
```
Comment (llettieri) [inline models/EventDate.ts:23]: Isn't something with dayjs also possible here? Looks bit complicated but idk if it's simpler with dayjs

Comment (nilshindermann) [reply models/EventDate.ts:23]: Done

**Review by llettieri — APPROVED** [review]

**Pull request merged**